model/src/w3iogomd.F90

```diff
@@ -2994,125 +3058,267 @@ SUBROUTINE W3IOGO ( INXOUT, NDSOG, IOTST, IMOD )
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PDIR:', PDIR(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 5 ) THEN
               WRITE ( NDSOG ) PSI(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PSI:', PSI(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 6 ) THEN
               WRITE ( NDSOG ) PWS(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PWS:', PWS(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 7 ) THEN
               WRITE ( NDSOG ) PTHP0(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PTHP0:', PTHP0(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 8  ) THEN
               WRITE ( NDSOG ) PQP(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PQP:', PQP(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 9  ) THEN
               WRITE ( NDSOG ) PPE(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PPE:', PPE(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 10 ) THEN
               WRITE ( NDSOG ) PGW(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PGW:', PGW(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 11 ) THEN
               WRITE ( NDSOG ) PSW(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PSW:', PSW(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 12 ) THEN
               WRITE ( NDSOG ) PTM1(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PTM1:', PTM1(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 13 ) THEN
               WRITE ( NDSOG ) PT1(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PT1:', PT1(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 14 ) THEN
               WRITE ( NDSOG ) PT2(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PT2:', PT2(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 15 ) THEN
               WRITE ( NDSOG ) PEP(1:NSEA,0:NOSWLL)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PEP:', PEP(1:NSEA,0:NOSWLL)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 16 ) THEN
               WRITE ( NDSOG ) PWST(1:NSEA)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PWST:', PWST(1:NSEA)
+#endif
             ELSE IF ( IFI .EQ. 4 .AND. IFJ .EQ. 17 ) THEN
               WRITE ( NDSOG ) PNR(1:NSEA)
+#ifdef W3_ASCII
+              WRITE ( NDSOA,* ) 'PNR:', PNR(1:NSEA)
+#endif
               !
               !     Section 5)
               !
             ELSE IF ( IFI .EQ. 5 .AND. IFJ .EQ. 1 ) THEN
               DO ISEA=1, NSEA
-                IX     = MAPSF(ISEA,1)
-                IY     = MAPSF(ISEA,2)
-                IF ( MAPSTA(IY,IX) .EQ. 1 ) THEN
+                IF ( UST(ISEA) .NE. UNDEF .AND. ASF(ISEA) .NE. UNDEF ) THEN
```

Comment:
@mickaelaccensi @MatthewMasarik-NOAA -- tagging you in the code as well where I just made the comment about where I think the answer changes are coming from.